huggingface_bot.py

```diff
@@ -0,0 +1,64 @@
+"""
+
+Bot that lets you talk to conversational models available on HuggingFace.
+
+"""
+from __future__ import annotations
+
+from typing import AsyncIterable
+
+from fastapi_poe import PoeBot
+from fastapi_poe.types import QueryRequest
+from huggingface_hub import InferenceClient
+from sse_starlette.sse import ServerSentEvent
+
+
+class HuggingFaceBot(PoeBot):
+    """This bot uses the HuggingFace Inference API.
+
+    By default, it uses the HuggingFace public Inference API, but you can also
+    use this class with a self hosted Inference Endpoint.
+    For more information on how to create a self hosted endpoint, see:
+    https://huggingface.co/blog/inference-endpoints
+
+    Arguments:
+        - model: either the name of the model (if you want to use the public API)
+        or a link to your hosted inference endpoint.
+
+    """
+
+    def __init__(self, model: str, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        if model is None:
```

Comment:
Instead of this explicit check, better to make this class a dataclass and let dataclasses take care of making a good constructor.